### DISCUSSION TITLE
There is a coding problem with a couple of programs...

### DISCUSSION THREAD

**Discussion by Phuxxi** [community/community#73228] [Programming Help]
### Select Topic Area

Bug

### Body

Hi all this is a simple problem that involves several parties.

First off Christian Boxdoerfer makes a truly exceptional search program for Linux called Fsearch.

Ryo Nakano   Reco  -  Record talks to remember the contents later

https://github.com/ryonakano/reco

Take photos and record videos with your webcam

https://webcamoid.github.io/

The problem I am now facing is that when I do a search for files - mp3's, pictures, videos, documents etc., using Fsearch, these programs that have NOTHING to do with any thing, have backed up a file NAME list - of everything on my hard drives... like 1.5 million files.

And Fsearch cannot be faulted, but after much looking I can't see any overt ways to cut specific directories from the search.

You see there are legitimate file name pathways, e.g. Home/Desktop/Elvis Movies/ etc. And then there are these files... 

Mostly these two: And there are copies of EVERY file NAME on the system....

/run/user/1000/doc/by-app/com.github.ryonakano.reco/ etc.

/run/user/1000/doc/by-app/io.github.webcamoid.Webcamoid/ etc.

So the programs have coding defects.

Can the people responsible for the apparently copied or replicated coding error please clean them up.

And Christian - if you could include a way to permanently lock out some pathways when doing a search.

Thanks to all.

Shane T. Hanson

**Comment by Phuxxi** [comment]
Like so.

![Idi Amin](https://github.com/community/community/assets/72900099/89ae649f-a356-4d6d-a4d3-255fed7d7319)

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Phuxxi ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.

**Comment by davea42** [comment]
The following may help. It explains what the /run directory is.


`https://www.linuxadictos.com/en/the-linux-run-directory.html`

On a system here I get
```
q3 533: df -k /run
Filesystem     1K-blocks  Used Available Use% Mounted on
tmpfs            3280100  1880   3278220   1% /run
```

which means things listed in /run are really in an in-memory
file system with links to real things.  The things are not in /run, only symlinks are  in /run.

So you really have just one copy, the one in Desktop.
Just figure out how to make Fsearch restrict its attention to $HOME.

I personally use catfish, a Linux app that by default restricts attention to one's own
home directory and things  in that or subdirectories.`  sudo apt-get install catfish`
I hope this helps.

**Comment by davea42** [comment]
Oh. And the solution to your issue is to, for example, do:

`fsearch $HOME  AMIN`

(There is usually not a good reason to be completely specific,  and with spaces in names 
it can be easier to  find what you want with a less specific request.  Kind of like
web searches, in a way)
